From 1fc6574a46bdc2cd1aec6414848d076a477ac343 Mon Sep 17 00:00:00 2001
From: Jay Cornwall <jay@jcornwall.me>
Date: Fri, 28 Aug 2015 11:25:39 -0500
Subject: [PATCH 1709/2094] drm/amdgpu: Enable BIF 5.0 atomic promotion to PCIe
 3.0 AtomicOps

Atomic operations to system memory are normally dropped by the BIF.
Some PCIe 3.0 platforms support atomic routing to the host for execution
by the CPU. To use this feature, atomic requests must be enabled in the
GPU's PCIe configuration space.

This is a no-op on platforms which cannot route the atomic request to
the host. In this instance, atomics would be dropped with or without
this change.

Signed-off-by: Jay Cornwall <jay@jcornwall.me>
Reviewed-by: Alex Deucher <alexander.deucher@amd.com>
Acked-by: John Bridgman <john.bridgman@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/gmc_v8_0.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/amd/amdgpu/gmc_v8_0.c b/drivers/gpu/drm/amd/amdgpu/gmc_v8_0.c
index f56b408..a83d32d 100644
--- a/drivers/gpu/drm/amd/amdgpu/gmc_v8_0.c
+++ b/drivers/gpu/drm/amd/amdgpu/gmc_v8_0.c
@@ -1144,6 +1144,10 @@ static int gmc_v8_0_sw_init(void *handle)
 		adev->vm_manager.enabled = true;
 	}
 
+	/* Allow BIF to recode atomics to PCIe 3.0 AtomicOps.
+	 */
+	pci_enable_atomic_request(adev->pdev);
+
 	return r;
 }
 
-- 
2.7.4

